packages/manager/src/features/CloudPulse/shared/CloudPulseRegionSelect.tsx

```diff
@@ -40,13 +52,33 @@ export const CloudPulseRegionSelect = React.memo(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [regions]);
 
+    // validate launchDrakly region_ids with the ids from the fetched 'all-regions'
+    const supportedRegions = React.useMemo<Region[] | undefined>(() => {
+      const resourceTypeFlag = flags.aclpResourceTypeMap?.find(
+        (item: CloudPulseResourceTypeMapFlag) =>
+          item.serviceType === serviceType
+      );
+
+      if (isNil(resourceTypeFlag?.supportedRegionIds)) {
```

Comment:
```suggestion
      if (!resourceTypeFlag?.supportedRegionIds) {
```
 Let's avoid the use of `isNil` here. The util comes from ramda, and we're trying to get rid of that package where we can because what it achieves can often be done in a simpler way.

With the optional chaining operator, if `supportedRegionIds` are undefined or null, the entire object should return `undefined` and the conditional will evaluate to true to return all regions.

Reply:
optional chaining character will not give correct result when the supportedRegionIds is an empty string, but to avoid the above util - isNil, i will use custom implementation and push. Thanks!